### PR TITLE
pull request for test cases and downloading exercises,content and topics

### PR DESCRIPTION
**Pull request by gwong89**

### CODE REVIEW

**Comment by aronasorman** [comment]
Merge conflict. Can you resolve it? There's instructions in this PR on how to resolve it.

**Comment by aronasorman** [comment]
Still has conflcits.

File: contentpacks/khanacademy.py [hunk 1]
```diff
@@ -79,8 +79,34 @@ def retrieve_language_resources(lang: str, version: str) -> LangpackResources:
     return LangpackResources(topic_data, content_data, exercise_data, subtitle_data, kalite_catalog, ka_catalog, dubbed_video_mapping)
 
 
-def retrieve_subtitles(videos: list, lang="en") -> dict:
-    return {}
+def retrieve_subtitles(videos: list, lang="en") -> list:
+    #videos => contains list of youtube ids
```
Comment (aronasorman) [inline contentpacks/khanacademy.py:83]: Rather than having this is a comment, make it a doctring instead.

File: tests/test_contentpacks.py [hunk 2]
```diff
@@ -19,6 +19,46 @@
 logging.basicConfig()
 logging.getLogger("vcr").setLevel(logging.DEBUG)
 
+class Test_retrieve_subtitles:
+    def test_incorrect_youtube_id(self):
+        incorrect_list = ["aaa"]
+        empty_list = retrieve_subtitles(incorrect_list)
+        test_list = []
+        assert type(empty_list) == list
+        assert len(empty_list) == 0
```
Comment (aronasorman) [inline tests/test_contentpacks.py]: You can summarize these last two asserts as `assert not empty_list`.

File: tests/test_contentpacks.py [hunk 3]
```diff
@@ -19,6 +19,46 @@
 logging.basicConfig()
 logging.getLogger("vcr").setLevel(logging.DEBUG)
 
+class Test_retrieve_subtitles:
+    def test_incorrect_youtube_id(self):
+        incorrect_list = ["aaa"]
+        empty_list = retrieve_subtitles(incorrect_list)
+        test_list = []
+        assert type(empty_list) == list
```
Comment (aronasorman) [inline tests/test_contentpacks.py]: Use `isinstance` instead.

File: tests/test_contentpacks.py [hunk 4]
```diff
@@ -19,6 +19,46 @@
...
+        assert type(filled_list) == list
+        assert len(filled_list) == 1
+        assert filled_list == test_list
+
+    def test_correct_and_incorrect_youtube_id(self):
+        mixed_list =  ["y2-uaPiyoxc", "asdadsafa"]
+        filled_list = retrieve_subtitles(mixed_list)
+        test_list = ["y2-uaPiyoxc"]
+        assert type(filled_list) == list
+        assert len(filled_list) == 1
```
Comment (aronasorman) [inline tests/test_contentpacks.py]: Read up on Python's truthiness idioms and then try to remove the redundancies in your assert statements.

File: contentpacks/khanacademy.py [hunk 5]
```diff
@@ -79,8 +79,34 @@ def retrieve_language_resources(lang: str, version: str) -> LangpackResources:
...
+    #return list of youtubeids that were downloaded
+    downloaded_videos = []
+    not_downloaded_videos = []
+    for youtube_id in videos:
+        request_url = "https://www.amara.org/api2/partners/videos/?format=json&video_url=http://www.youtube.com/watch?v=%s" % (
+        youtube_id)
+        response =  requests.get(request_url)
```
Comment (aronasorman) [inline contentpacks/khanacademy.py]: The idiom in the requests library is to use `raise_for_status` to check if we got a nonsucessful error code: http://docs.python-requests.org/en/latest/user/quickstart/

File: contentpacks/khanacademy.py [hunk 6]
```diff
@@ -79,8 +79,34 @@ def retrieve_language_resources(lang: str, version: str) -> LangpackResources:
...
+        else:
+            amara_id = content["objects"][0]["id"]
+            subtitle_download_uri = "https://www.amara.org/api/videos/%s/languages/%s/subtitles/?format=vtt" %(amara_id, lang)
+            
+            try:
+                response_code = urllib.request.urlopen(subtitle_download_uri)
+
+            except urllib.error.HTTPError:
+                continue   
+            FILE_DIR = os.path.join(os.getcwd(), "build", "subtitles", lang)
```
Comment (aronasorman) [inline contentpacks/khanacademy.py]: Don't make variables all caps if they aren't constants.

**Comment by aronasorman** [comment]
Merge conflict.

File: contentpacks/khanacademy.py (hunk 5 above)
Comment (gwong89) [inline contentpacks/khanacademy.py]: are you referring for line 90 or 100? 90 will always return a 200 regardless of a bad youtube id, unless the server is down then I don't see a point using it over there, unless you meant 100

**Pull request merged**